api/v1alpha4/types.go

```diff
@@ -116,6 +116,32 @@ type SubnetFilter struct {
 	NotTagsAny      string `json:"notTagsAny,omitempty"`
 }
 
+type PortOpts struct {
+	// ID of the OpenStack network on which to create the port. If unspecified, create the port on the default cluster network.
+	NetworkID string `json:"networkId,omitempty"`
+	// Used to make the name of the port unique. If unspecified, instead the 0-based index of the port in the list is used.
+	NameSuffix   string `json:"nameSuffix,omitempty"`
+	Description  string `json:"description,omitempty"`
+	AdminStateUp *bool  `json:"adminStateUp,omitempty"`
+	MACAddress   string `json:"macAddress,omitempty"`
+	// Specify pairs of subnet and/or IP address. These should be subnets of the network with the given NetworkID.
+	FixedIPs       []FixedIP `json:"fixedIps,omitempty"`
+	TenantID       string    `json:"tenantId,omitempty"`
+	ProjectID      string    `json:"projectId,omitempty"`
+	SecurityGroups *[]string `json:"securityGroups,omitempty"`
+
```

Comment:
Could we include allowedAddressPairs as well? 

Examples of Implementation:
https://github.com/openshift/cluster-api-provider-openstack/blob/3024c78c7026e804f8c4b5765c652110f6df4d7a/pkg/apis/openstackproviderconfig/v1alpha1/types.go#L211
https://github.com/openshift/cluster-api-provider-openstack/blob/3024c78c7026e804f8c4b5765c652110f6df4d7a/pkg/apis/openstackproviderconfig/v1alpha1/types.go#L226-L229
https://github.com/openshift/cluster-api-provider-openstack/blob/3024c78c7026e804f8c4b5765c652110f6df4d7a/pkg/cloud/openstack/clients/machineservice.go#L348-L366

Reply:
Yes, of course, I'll add `allowedAddressPairs` too.